components/layout_2020/display_list/mod.rs

```diff
@@ -414,6 +432,10 @@ impl Fragment {
             rect.size.height = Length::new(font_metrics.strikeout_size.to_nearest_pixel(dppx));
             self.build_display_list_for_text_decoration(fragment, builder, &rect, &color);
         }
+
+        if !shadows.0.is_empty() {
+            builder.wr().pop_all_shadows();
```

Comment:
Why is this needed?

Reply:
In WebRender you push shadows and pop them. All display items that you add to the display list between a push and a pop get a shadow and are shadowed together to properly preserve layering.